### PR TITLE
Implement Serialized Entity Pattern

### PR DESCRIPTION
**Pull request by JurenXu**
### Serialized Entity Pattern
+ Solves #1275 

### Files added:
- App.java
- Country.java
- CountryDao.java
- CountrySchemaSql.java
- CountryTest.java
- README.md
- serialize-entity.urm.puml
- pom.xml (inherit from parent pom.xml) 

### Update files:
- pom.xml (parent pom.xml, add 1 module called serialized-entity)

### CODE REVIEW

**Comment by JurenXu** [comment]
Hi, I copied pom.xml from the servant module for convenience and caused some errors. 
I accidentally changed the servant's pom.xml code while I was fixing these issues, but I've now restored it.
This may have caused some problems for you in reviewing the code, and I apologize for that

**Comment by JurenXu** [comment]
@iluwatar Please review this pull request

File: serialized-entity/README.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
+---
+layout: pattern
+title: Serialized Entity Pattern
+folder: serialized-entity
+permalink: /patterns/serialized-entity/
+categories: Behavioral
+language: en
+tags:
+- Data access
+---
```
Comment (iluwatar) [inline serialized-entity/README.md:10]: Please check the updated YAML frontmatter requirements in https://github.com/iluwatar/java-design-patterns/wiki/01.-How-to-contribute

File: serialized-entity/src/main/java/com/iluwatar/serializedentity/App.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import java.sql.SQLException;
+import javax.sql.DataSource;
+import lombok.extern.slf4j.Slf4j;
+import org.h2.jdbcx.JdbcDataSource;
+
+
+/**
+ * Serialized Country objects and store them into database.
+ * Read data from database and deserialize them and reconstruct to Country objects.
+ */
```
Comment (iluwatar) [inline serialized-entity/src/main/java/com/iluwatar/serializedentity/App.java:35]: Describe the pattern and explain how this example implements it

File: serialized-entity/src/main/java/com/iluwatar/serializedentity/App.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            "Arabic"
+    );
+
+    final var serializedChina = new CountrySchemaSql(China, dataSource);
+    final var serializedUnitedArabEmirates = new CountrySchemaSql(UnitedArabEmirates, dataSource);
+    serializedChina.insertCountry();
+    serializedUnitedArabEmirates.insertCountry();
+
+    serializedChina.selectCountry();
+    serializedUnitedArabEmirates.selectCountry();
```
Comment (iluwatar) [inline serialized-entity/src/main/java/com/iluwatar/serializedentity/App.java:76]: We need comments explaining what is going on

File: serialized-entity/README.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+      preparedStatement.setInt(1, country.getCode());
+
+      try (ResultSet rs = preparedStatement.executeQuery()) {
+        if (rs.next()) {
+          Blob countryBlob = rs.getBlob("country");
+          ByteArrayInputStream baos = new ByteArrayInputStream(countryBlob.getBytes(1, (int) countryBlob.length()));
+          ObjectInputStream ois = new ObjectInputStream(baos);
+          country = (Country) ois.readObject();
+          System.out.println(country);
```
Comment (iluwatar) [inline serialized-entity/README.md:118]: Use logger

File: serialized-entity/src/main/java/com/iluwatar/serializedentity/CountrySchemaSql.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+      preparedStatement.setInt(1, country.getCode());
+
+      try (ResultSet rs = preparedStatement.executeQuery()) {
+        if (rs.next()) {
+          Blob countryBlob = rs.getBlob("country");
+          ByteArrayInputStream baos = new ByteArrayInputStream(countryBlob.getBytes(1, (int) countryBlob.length()));
+          ObjectInputStream ois = new ObjectInputStream(baos);
+          country = (Country) ois.readObject();
+          System.out.println(country);
```
Comment (iluwatar) [inline serialized-entity/src/main/java/com/iluwatar/serializedentity/CountrySchemaSql.java:112]: Use logger

**Review by iluwatar — CHANGES_REQUESTED** [review]
Looks pretty good already. Left a couple of comments to resolve.

File: serialized-entity/src/main/java/com/iluwatar/serializedentity/CountrySchemaSql.java (hunk 5 above)
Comment (JurenXu) [reply serialized-entity/src/main/java/com/iluwatar/serializedentity/CountrySchemaSql.java]: I have resolved all changes you required! Could you check it again?

Comment (JurenXu) [reply serialized-entity/src/main/java/com/iluwatar/serializedentity/CountrySchemaSql.java]: > Use logger

have resolved all changes you required! Could you check it again?

Comment (JurenXu) [reply serialized-entity/src/main/java/com/iluwatar/serializedentity/CountrySchemaSql.java]: > Use logger

I have resolved all changes you required! Could you check it again?

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Looks good! Thank you for the contribution 🎉

@all-contributors please add @JurenXu for code

**Pull request merged**